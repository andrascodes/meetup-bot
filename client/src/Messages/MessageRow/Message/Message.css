.TimeIndicator {
  font-size: 12px;
  margin-bottom: 0.5em;
  color: rgba(0, 0, 0, .40);
}

.Indicators {
  font-size: 12px;
  margin-bottom: 0.5em;
  color: rgba(0, 0, 0, .40);
}