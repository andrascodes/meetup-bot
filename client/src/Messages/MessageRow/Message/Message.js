import React from 'react'

import { formatMessageTimestamp } from '../../../lib/dateManipulation'
import { SentimentIndicator as Sentiment } from './SentimentIndicator'
import { ErrorIndicator as Error } from './ErrorIndicator'
import { Bubble } from './Bubble'

import './Message.css'

export const Message = props => {

  return (
    <div>
      <div className="TimeIndicator text-center">
        {props.direction === 'incoming' ?
          `Arrived at ` : `Sent at `
        }
        {formatMessageTimestamp(props.timestamp)}
      </div>

      <Bubble type={props.type}
        text={props.text}
        direction={props.direction} 
        message={props.message} 
        response={props.response}
        error={props.error}
      />
      
      <div className="row Indicators">
        <div className="col-lg-6 col-md-6 col-sm-6 col-xs-6 text-left">
          <Sentiment value={props.sentiment} />
        </div>
        <div className="col-lg-6 col-md-6 col-sm-6 col-xs-6 text-right">
          <Error value={props.error} response={props.response}/>
        </div>
      </div>
    </div>
  )
}