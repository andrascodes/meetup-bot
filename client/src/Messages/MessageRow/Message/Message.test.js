import React from 'react'
import renderer from 'react-test-renderer'

import { Message } from './Message'

describe('Message', () => {

  test('renders correctly', () => {
    const tree = renderer.create(
      <Message
        timestamp={'2017-04-10T15:16:55.834Z'}
        type={'text'}
        direction={'incoming'} 
        message={{
          message: {
            text: 'Incoming text message'
          }
        }}
        text={'Text: Incoming text message'}
        response={{
          error: {
            code: 200,
          }
        }}
        error={'OAuthException'}
        sentiment={'neutral'}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

})