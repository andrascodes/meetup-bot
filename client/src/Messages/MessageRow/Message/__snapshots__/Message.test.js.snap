exports[`Message renders correctly 1`] = `
<div>
  <div
    className="TimeIndicator text-center">
    Arrived at 
    17:16:55
  </div>
  <div>
    <div
      className="Incoming Bubble ErrorBubble"
      data-tip="React tooltip">
      <div
        className="TextMessage">
        Incoming text message
      </div>
    </div>
    <div
      className="ErrorTooltip">
      <div
        className="__react_component_tooltip place-top type-dark "
        data-id="tooltip">
        
      </div>
    </div>
  </div>
  <div
    className="row Indicators">
    <div
      className="col-lg-6 col-md-6 col-sm-6 col-xs-6 text-left">
      <div
        className="Sentiment">
        Sentiment: 
        <span
          className="Neutral">
          neutral
        </span>
        <i
          className="fa fa-check Neutral Icon" />
      </div>
    </div>
    <div
      className="col-lg-6 col-md-6 col-sm-6 col-xs-6 text-right">
      <div
        className="ErrorIndicator">
        Error: 
        <span
          className="ErrorStyle">
          OAuthException #200
        </span>
        <i
          className="fa fa-times ErrorStyle Icon" />
      </div>
    </div>
  </div>
</div>
`;
